### PR TITLE
Minor derive_lq improvment

### PR DESCRIPTION
**Pull request by Aeilert**

### CODE REVIEW

File: R/gd_compute_pip_stats_lq.R [hunk 1]
```diff
@@ -165,7 +165,9 @@ derive_lq <- function(x, A, B, C) {
   alpha <- (B^2) - (4 * A)
   beta <- (2 * B * e) - (4 * C) # C is called D in original paper, but C in Datt paper
   tmp <- (alpha * x^2) + (beta * x) + (e^2)
-  tmp <- ifelse(tmp < 0, 0, tmp) # Why would we set tmp to 0? It would still fail: division by 0.
+  tmp <- if (!is.na(tmp)) {
+    if (tmp < 0) 0 else tmp # Why would we set tmp to 0? It would still fail: division by 0.
+  }
```
Comment (tonyfujs) [inline R/gd_compute_pip_stats_lq.R:170]: This is faster than the original code

**Review by tonyfujs — COMMENTED** [review]
Hey @Aeilert 
I'm not sure about this change. It makes part of the code faster, but other part slower... I don't know what the overall impact  will be. Any thought?

File: R/gd_compute_pip_stats_lq.R (hunk 1 above)
Comment (Aeilert) [reply R/gd_compute_pip_stats_lq.R:170]: > Hey @Aeilert
> I'm not sure about this change. It makes part of the code faster, but other part slower... I don't know what the overall impact will be. Any thought?
> 

Hey @tonyfujs . 
Yep. There is a trade-off here. But the idea is that `derive_lq` is used a lot for the poverty calculations in the API, while `sd_compute_dist_stats()` is only used in the pipeline. What do you think? I was thinking that is a trade-off we would be  willing to make.

Comment (tonyfujs) [reply R/gd_compute_pip_stats_lq.R:170]: Thanks @Aeilert 
Oh right! You actually already mentioned this, and I forgot... Indeed. Sounds like a sensible trade-off.

**Pull request merged**